apps/console/src/features/applications/components/settings/attribute-management/attribute-selection.tsx

```diff
@@ -308,16 +315,26 @@ export const AttributeSelection: FunctionComponent<AttributeSelectionPropsInterf
             if (claimConfigurations?.dialect === "CUSTOM") {
                 const initialClaimMappingList: ExtendedClaimMappingInterface[] = [];
                 claimConfigurations.claimMappings.map((claim) => {
+                    if (!claim || !claim.localClaim) {
+                        return;
+                    }
+                    if (claim.localClaim.uri === defaultSubjectAttribute &&
+                        isSubjectClaimSetToDefaultWithoutMapping()) {
+                        return;
+                    }
                     const claimMapping: ExtendedClaimMappingInterface = {
                         addMapping: true,
-                        applicationClaim: claim?.applicationClaim,
+                        applicationClaim: claim.applicationClaim,
```

Comment:
Do we need to remove these undefined checks?

Reply:
With the below section added(line 318), we no longer need to have those undefuned checks.
```
if (!claim || !claim.localClaim) {
         return;
}
```